server/src/Subscription.hs

```diff
@@ -77,18 +76,16 @@ unsubscribe path subid (SubscriptionTree here inner) =
 
 -- Invoke f for all subscribers to the path. The subscribers get passed the
 -- subvalue at the path that they are subscribed to.
-broadcast :: (conn -> Value -> IO ()) -> [Text] -> Value -> SubscriptionTree id conn -> IO ()
-broadcast f path value tree =
-  -- We broadcast concurrently since all updates are independent of each other
-  Async.mapConcurrently_ (uncurry f) notifications
+broadcast :: (state -> Value -> IO ()) -> [Text] -> Value -> SubscriptionTree id state -> IO ()
+broadcast f path value tree = mapM_ (uncurry f) notifications
   where notifications = broadcast' path value tree
 
 -- Like broadcast, but return a list of notifications rather than invoking an
 -- effect on each of them.
-broadcast' :: [Text] -> Value -> SubscriptionTree id conn -> [(conn, Value)]
+broadcast' :: [Text] -> Value -> SubscriptionTree id state -> [(state, Value)]
 broadcast' = \path value tree -> execWriter $ loop path value tree
   where
-  loop :: [Text] -> Value -> SubscriptionTree id conn -> Writer [(conn, Value)] ()
+  loop :: [Text] -> Value -> SubscriptionTree id state -> Writer [(state, Value)] ()
```

Comment:
I suppose we rarely have updates that affect a lot of subscribers at once, otherwise, this could also be an issue due to using `Writer`. For one, even the "strict" writer is still lazy in the value, so it's building large thunks, and the monoid used here (`[]`) has `O(n)` append.

Not an issue to be solved in this PR, but something to keep in mind.

Reply:
Good point. We should keep track of this somewhere. @DiegoDiverio can you open an issue for this and just note it down as a potential future improvement. You can then just put a comment here in the code and link to that issue.

Reply:
More recent `transformers` / `mtl` have a CPSd version of `Writer` which is less space-leaky. Maybe that's worth using since it's just an import change?

Reply:
If you're referring to https://hackage.haskell.org/package/mtl-2.3.1/docs/Control-Monad-Writer-CPS.html, I think that's a good idea, we just need a version bump for `mtl` (we have `< 2.3` as a version restriction at the moment) and the import change.